File: Recursion/PowerOf.c
#include <stdio.h>

int powerof(int b,int p){
    if(p == 1) {return b;}
    
    int t = powerof(b,p/2) * powerof(b,p/2);
    
    if(p % 2 == 0){
        return t;
    }
    return t * b;
}

int main(){
    int b,p;
    
    printf("Enter a base: ");
    scanf("%d",&b);
    printf("Enter a power: ");
    scanf("%d",&p);
    
    int result = powerof(b,p);
    printf("\n%d^%d = %d",b,p,result);
}

File: Recursion/PowerOf.py
def power(b,p):
    if p == 1: return b
    
    t = power(b,int(p/2)) * power(b,int(p/2))
    
    if(p % 2 == 0):
        return t
    else:
        return t * b

print("Enter a base: ")
b = int(input())
print("Enter a power: ")
p = int(input())

print(f"\n{b}^{p} = {power(b,p)}")